File: server/medicore.apps/src/model/articleImageModel.ts
import { IEntityBase } from './IEntityBase';
import { Type } from 'class-transformer';
import { ArticleModel, ImageModel } from '../model';

export class ArticleImageModel implements IEntityBase {
    id: number;
    articleId: number;
    imageId: number;
    @Type(() => ArticleModel)
    article: ArticleModel;
    @Type(() => ImageModel)
    image: ImageModel;

    constructor() {
        this.id = 0;
        this.imageId = 0;
        this.article = null;
        this.image = null;
    }
}

File: server/dna.core/dna.core.service/Models/ArticleImageModel.cs
using dna.core.service.Models.Abstract;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace dna.core.service.Models
{
    public class ArticleImageModel : IModelBase
    {
        public int Id { get; set; }

        [Required]
        public int ArticleId { get; set; }
        [Required]
        public int ImageId { get; set; }

     
        public virtual ArticleModel Article { get; set; }
        public virtual ImageModel  Image { get; set; }
    }
}
